## Big Data: Checkpoint 4

In the following Exercieses, you'll be working with the code snippet below:

In [ ]:
%%timeit
x = da.random.random((10000, 10000), chunks=(1000, 1000))
y = x + x.T
z = y[::2, 5000:].mean(axis=1)
z.compute()

Please submit your solutions to the following tasks as a link to your jupyter notebook on Github.

## 1. Change the code above by setting `chunks=(250, 250)`. How long does it take to run?

In [2]:
import dask.array as da
import numpy as np

In [4]:
%%timeit
x = da.random.random((10000, 10000), chunks=(1000, 1000))
y = x + x.T
z = y[::2, 5000:].mean(axis=1)
z.compute()

1.38 s ± 94.4 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)


In [5]:
)'.%%timeit
x = da.random.random((10000, 10000), chunks=(250, 250))
y = x + x.T
z = y[::2, 5000:].mean(axis=1)
z.compute()

4.51 s ± 176 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)


It took around 1.23 seconds or 1230 milliseconds.


## 2. Now, set the parameter to `chunks=(500, 500)`.How long does it take to run? Does this one or the previous one run quickly? Why?

In [6]:
%%timeit
x = da.random.random((10000, 10000), chunks=(500, 500))
y = x + x.T
z = y[::2, 5000:].mean(axis=1)
z.compute()

1.93 s ± 141 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)


This run took about 1.93 seconds, which is approximately half the time it took for the task chunks sized at (250,250). A dask array that is split into smaller chunks means that there will be more chunks to handle and therefore a higher computation load. A dask array split into larger chunks will have a smaller computation load, and will finish sooner given the available memory is large enough to handle the chunk size.  